### PR TITLE
Wasm 1.0

### PR DESCRIPTION
**Pull request by EItanya**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/1868

### CODE REVIEW

File: projects/gloo/pkg/upstreams/consul/upstream_client_test.go [hunk 1]
```diff
@@ -159,7 +159,8 @@ var _ = Describe("ConsulClient", func() {
 			})
 		})
 
-		Context("a transient error occurs while contacting the Consul agent", func() {
+		// TODO(marco): temporarily disable, want to see if there are other failures in CI while I investigate
+		XContext("a transient error occurs while contacting the Consul agent", func() {
```
Comment (rickducott) [inline projects/gloo/pkg/upstreams/consul/upstream_client_test.go]: Remove

File: changelog/v1.2.0/helm-3.yaml [hunk 2]
```diff
@@ -40,8 +40,13 @@ changelog:
     and should therefore be used only when strictly necessary.
     Not being considered part of a release, hooks will not be cleaned up when running `helm uninstall`. The documented
     ways to get around this will not work for us though: we can't use `hook-delete-policy` since we have multiple hooks
+<<<<<<< HEAD
```
Comment (rickducott) [inline changelog/v1.2.0/helm-3.yaml]: merge conflict

File: Makefile [hunk 3]
```diff
@@ -358,7 +381,8 @@ HELM_DIR := install/helm
 INSTALL_NAMESPACE ?= gloo-system
 
 .PHONY: manifest
-manifest: prepare-helm install/gloo-gateway.yaml install/gloo-ingress.yaml install/gloo-knative.yaml update-helm-chart
+manifest: prepare-helm install/gloo-gateway.yaml install/gloo-ingress.yaml \
+ 		install/gloo-knative.yaml update-helm-chart
```
Comment (ilackarms) [inline Makefile]: what was the reason for the change?

Comment (EItanya) [reply Makefile]: This lines were so long they stretched off of my screen. This makes them readable. Does it change behavior

File: pkg/utils/protoutils/utils.go [hunk 4]
```diff
@@ -44,16 +43,15 @@ func MarshalStructEmitZeroValues(m proto.Message) (*structpb.Struct, error) {
...
 	if structuredData == nil {
 		return NilStructError
 	}
 	strData, err := jsonpbMarshaler.MarshalToString(structuredData)
 	if err != nil {
 		return err
 	}
-	data := []byte(strData)
-	return json.Unmarshal(data, into)
+	return jsonpb.UnmarshalString(strData, into)
```
Comment (ilackarms) [inline pkg/utils/protoutils/utils.go]: can we change the behavior of this function safely? seems like this function was specifically designed to b used with non-proto objects (converts a proto struct into generic JSON, then generic json to generic go struct)

Comment (EItanya) [reply pkg/utils/protoutils/utils.go]: This function wasn't being used at all. I created it as a utility to deal with non-gogo structs. It didn't work as implemented before. Not 100% sure why yet, but didn't have time to investigate

File: projects/envoyinit/wasm/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,6 @@
+FROM quay.io/solo-io/envoy-gloo-wasm:0.2.0
+
+COPY envoyinit-linux-amd64 /usr/local/bin/envoyinit
+
+ENTRYPOINT ["/usr/bin/dumb-init", "--", "/usr/local/bin/envoyinit"]
+CMD []
```
Comment (ilackarms) [inline projects/envoyinit/wasm/Dockerfile]: why have a `CMD` and an `ENTRYPOINT` ?

File: projects/envoyinit/wasm/main.go [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+	return "/etc/envoy/envoy.yaml"
+}
+
+func outputCfg() string {
+	maybeConf := os.Getenv("OUTPUT_CONF")
+	if maybeConf != "" {
+		return maybeConf
+	}
+	return "/tmp/envoy.yaml"
+}
```
Comment (ilackarms) [inline projects/envoyinit/wasm/main.go]: this looks like a copy-paste of `envoyinit/cmd/main.go`

what purpose does this file provide?

File: pkg/utils/protoutils/utils.go (hunk 4 above)
Comment (ilackarms) [reply pkg/utils/protoutils/utils.go]: maybe a comment above the function to indicate how it's used would be helpful

File: projects/gloo/pkg/upstreams/consul/upstream_client_test.go (hunk 1 above)
Comment (ilackarms) [reply projects/gloo/pkg/upstreams/consul/upstream_client_test.go]: bump

File: projects/gloo/pkg/translator/listener.go [hunk 7]
```diff
@@ -103,7 +103,7 @@ func (t *translatorInstance) computeListenerFilters(params plugins.Params, liste
 
 	// return if listener type != http || no virtual hosts
 	httpListener, ok := listener.ListenerType.(*v1.Listener_HttpListener)
-	if !ok || len(httpListener.HttpListener.VirtualHosts) == 0 {
```
Comment (yuval-k) [inline projects/gloo/pkg/translator/listener.go]: what's the purpose of this?

**Comment by yuval-k** [comment]
also, maybe point out in the docs somewhere that this feature is experimental

**Pull request merged**